core/src/OpenACC/Kokkos_OpenACC_ParallelFor_MDRange.hpp

```diff
@@ -30,10 +30,23 @@ void OpenACCParallelForMDRangePolicy(OpenACCCollapse, OpenACCIterateLeft,
                                      OpenACCMDRangeBegin<2> const& begin,
                                      OpenACCMDRangeEnd<2> const& end,
                                      int async_arg) {
-  int begin1 = begin[1];
-  int end1   = end[1];
-  int begin0 = begin[0];
-  int end0   = end[0];
+  auto begin1 = begin[1];
+  auto end1   = end[1];
+  auto begin0 = begin[0];
+  auto end0   = end[0];
+#if defined(KOKKOS_ENABLE_OPENACC_COLLAPSE_MDRANGE_LOOPS)
+  auto dim1  = end1 - begin1;
+  auto dim0  = end0 - begin0;
+  auto nIter = dim1 * dim0;
+// clang-format off
+#pragma acc parallel loop gang vector copyin(functor) async(async_arg)
+  // clang-format on
+  for (auto m = 0; m < nIter; ++m) {
```

Comment:
I appreciate the sentiment of the "auto" but `m` is going to be `int` this way, and if nIter turns out to be unsigned you still get a warning.

Reply:
Updated.